packages/ember-toucan-core/src/components/form/controls/autocomplete.ts

```diff
@@ -52,39 +48,31 @@ export interface ToucanFormAutocompleteControlComponentSignature<
      * Called when the user makes a selection.
      * It is called with the selected option (derived from `@options`) as its only argument.
      */
-    onChange?: (option: unknown) => void;
+    onChange?: (selected: string | null) => void;
 
     /**
      * The function called when a user types into the textbox, typically used to write custom filtering logic.
+     *
+     * @return An array of options to be shown in the popover.
      */
-    onFilter?: (input: string) => OPTION[];
+    onFilter?: (value: string) => string[];
```

Comment:
@ynotdraw Allowing the consumer to specify an array of filtered options means that what's displayed in the popover can diverge totally from the `@options` argument. It could be a totally different set of options—a whole different array—which means the consumer has two ways to specify options and which set is rendered will depend on whether the user has filtered.

Reply:
I can dig it. After noodling it a bit, the only thing that's a little off to me is that we'd be asking the consumer to do something imperative (add an event listener) for what previously had API support. The consumer would also have to deal with `event.target` and do an assertion on it before grabbing its `value`.

Reply:
> After noodling it a bit, the only thing that's a little off to me is that we'd be asking the consumer to do something imperative (add an event listener) for what previously had API support. The consumer would also have to deal with `event.target` and do an assertion on it before grabbing its `value`, eh?

Yeah, good point.  That could be a downside to folks if they use `@onFilter` frequently.

> Another option might be to give the consumer a filtering callback that's called with each iteration of `@options`. Or what, I think, @simonihmig suggested, where `@onFilter` simply calls Array.prototype.filter.

This is true!  We could do as Simon suggested here as the default to have _some_ baked in filtering support.  Then anything more advanced, folks could use the modifier and adjust `@options` possibly?

Reply:
I'm cool with `Array.prototype.filter`. 

What do you have in mind for the advanced case? I ask because, if we don't have a use-case in mind, then we might forgo suggesting it in the documentation as an alternative. Otherwise, if we have a reasonable advanced use-case, then I'll go ahead and document the possibility of the modifier.

Reply:
Awesome, sounds good to me.

> What do you have in mind for the advanced case? I ask because, if we don't have a use-case in mind, then we might forgo suggesting it in the documentation as an alternative.

Maybe we can chat about this this week but punt on it for now?